### PR TITLE
Add operators for Paddle

### PR DESCRIPTION
**Pull request by Wong4j**
The following ops are added:

- rmsnorm_fwd, rmsnorm_bwd, rmsnorm_fwd_fp8
- scaled (upper triang) (masked) softmax  fwd+bwd
- fused_attn_fwd_qkvpacked, fused_attn_fwd_kvpacked, fused_attn_bwd_qkvpacked, fused_attn_bwd_kvpacked

(To avoid merge conflict, I create my git branch based on @Tom-Zheng 's [PR262](https://github.com/NVIDIA/TransformerEngine/pull/262).  Please merge his PR first and then this one)

### CODE REVIEW

**Comment by ksivaman** [comment]
@Wong4j there are some conflicts due to #262

**Comment by Wong4j** [comment]
> @Wong4j there are some conflicts due to #262

Conflicts resolved.

**Comment by Wong4j** [comment]
CC @jeng1220  @Tom-Zheng

**Comment by Wong4j** [comment]
Hi @timmoon10 , would you please take a look?

File: transformer_engine/paddle/csrc/custom_ops.cu [hunk 1]
```diff
@@ -5,10 +5,83 @@
...
+    } else if (type == paddle::DataType::FLOAT16) {
+        fill_value<<<(size + block_size - 1) / block_size, block_size, 0, stream>>>(
+            tensor.data<half>(), size, static_cast<half>(0));
+    } else if (type == paddle::DataType::BFLOAT16) {
+        fill_value<<<(size + block_size - 1) / block_size, block_size, 0, stream>>>(
+            tensor.data<half>(), size, static_cast<half>(0));
+    } else {
+        NVTE_CHECK(false, "Unsupported data type");
+    }
+}
```
[the remark this answers is not in the document]
Comment (Wong4j) [reply transformer_engine/paddle/csrc/custom_ops.cu]: Paddle dost not provide these c++ functions currently. So, I do memset on the python side using paddle.full().
See the changes in the latest commit. https://github.com/NVIDIA/TransformerEngine/pull/285/commits/b8724e93a7e0a319b035cec09df4c578aa430143

File: transformer_engine/paddle/csrc/custom_ops.cu [hunk 2]
```diff
@@ -337,6 +410,609 @@ std::vector<paddle::Tensor> te_layernorm_bwd(const paddle::Tensor &dz, const pad
...
+                                                    scale_factor, input.stream());
+
+    return {softmax_results};
+}
+
+void te_scaled_upper_triang_masked_softmax_backward(paddle::Tensor &output_grads,  // NOLINT
+                                                    const paddle::Tensor &softmax_results,
+                                                    float scale_factor) {
+    NVTE_CHECK(output_grads.shape().size() == 3, "expected 4D tensor");
+    NVTE_CHECK(softmax_results.shape().size() == 3, "expected 4D tensor");
```
[the remark this answers is not in the document]
Comment (Wong4j) [reply transformer_engine/paddle/csrc/custom_ops.cu]: Done. https://github.com/NVIDIA/TransformerEngine/pull/285/commits/b8724e93a7e0a319b035cec09df4c578aa430143

File: tests/paddle/test_operators.py [hunk 3]
```diff
@@ -2,11 +2,13 @@
 #
 # See LICENSE for license information.
 """Test TE operators"""
-
+import struct
```
Comment (jeng1220) [inline tests/paddle/test_operators.py:5]: need to follow alphabet order. @Wong4j,  check the PRD.

**Review by timmoon10 — APPROVED** [review]

**Pull request merged**